### PR TITLE
fix some bugs about auto_col

### PR DESCRIPTION
**Pull request by jianwan0214**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #5530 #7508

## What this PR does / why we need it:

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Review by nnsgmsone — COMMENTED** [review]
Do you have the performance comparison data before and after this pr

File: pkg/sql/colexec/auto_incr.go [hunk 1]
```diff
@@ -249,35 +249,71 @@ func updateBatchImpl(ctx context.Context, ColDefs []*plan.ColDef, bat *batch.Bat
...
+	ret, err := rel.Ranges(param.ctx, expr)
 	if err != nil {
-		return 0, 0, err
+		return 0, 0, nil, err
 	}
 	switch {
 	case len(ret) == 0:
 		if rds, err = rel.NewReader(param.ctx, 1, nil, nil); err != nil {
-			return 0, 0, err
+			return 0, 0, nil, err
```
Comment (nnsgmsone) [inline pkg/sql/colexec/auto_incr.go:303]: where is the expr

**Comment by nnsgmsone** [comment]
Use the interface when manipulating batch instead of manipulating its variables directly.

**Review by nnsgmsone — APPROVED** [review]

**Comment by nnsgmsone** [comment]
bvt failed....

**Review by aunjgr — APPROVED** [review]

**Pull request merged**